components/x-interaction/src/concerns/register-component.js

```diff
@@ -1,9 +1,15 @@
 const registeredComponents = {}
 
-export function registerComponent(component) {
-	registeredComponents[component.wrappedDisplayName] = component
+export function registerComponent(component, name) {
+	const xInteractionName = Symbol('x-interaction-name')
+	component[xInteractionName] = name
+	registeredComponents[name] = component
 }
 
-export function getComponent(name) {
+export function getComponent(component) {
```

Comment:
this function is used by `hydrate` to get a component by its name, but here it seems to be taking a component as an argument. i think this needs to be split into two functions, `getComponent` which is the same as previously and `getComponentName` to replace `get-component-name.js` and return the component's name from the hidden property